framework/src/partitioner/LibmeshPartitioner.C

```diff
@@ -19,21 +19,24 @@
 #include "libmesh/metis_partitioner.h"
 #include "libmesh/hilbert_sfc_partitioner.h"
 #include "libmesh/morton_sfc_partitioner.h"
+#include "libmesh/subdomain_partitioner.h"
 
 template<>
 InputParameters validParams<LibmeshPartitioner>()
 {
   InputParameters params = validParams<MoosePartitioner>();
-  MooseEnum partitioning("metis=-2 parmetis=-1 linear=0 centroid hilbert_sfc morton_sfc");
+  MooseEnum partitioning("metis=-2 parmetis=-1 linear=0 centroid hilbert_sfc morton_sfc subdomain_partitioner");
   params.addRequiredParam<MooseEnum>("partitioner", partitioning, "Specifies a mesh partitioner to use when splitting the mesh for a parallel computation.");
   MooseEnum direction("x y z radial");
   params.addParam<MooseEnum>("centroid_partitioner_direction", direction, "Specifies the sort direction if using the centroid partitioner. Available options: x, y, z, radial");
+  params.addParam<std::vector<std::vector<SubdomainName> > >("blocks", "Block is seperated by ;, and partition mesh block by block. ");
```

Comment:
For example, `blocks = '1 2 3 4; 1001 1002 1003 1004'`, we use parmetis or other partitioners to partition the mesh consisting of blocks 1, 2, 3 and 4, and then use the same manner for blocks 1001, 1002, 1003 and 1004.